### ISSUE TITLE
E3033: String Length does not account for `resolve` strings such as SSM and SecretsManager 

### ISSUE THREAD

**Issue by BadAsstronaut** [aws-cloudformation/cfn-lint#2280]
*cfn-lint version: (`cfn-lint --version`)* 0.59.0

*Description of issue.*
Rules for string length should ignore `resolve` strings even if they do not meet length requirements.

Example:
```yaml
  SamlProvider:
    Type: AWS::IAM::SAMLProvider
    Properties:
      SamlMetadataDocument: '{{resolve:secretsmanager:client-vpn.saml-metadata}}'
```

Running cfn-lint gives the issue:  `[cfn-lint] E3033: String has to have length between 1000 and 10000000 at Resources...`
